### PR TITLE
Add support for Literal types

### PR DESCRIPTION
**Pull request by brentyi**
Hello,

Just added a special rule for the `Literal` annotation, which infers a type and populates argparse's `choices=` parameter. Getting something like this merged would make my life a bit easier!

Example from the test cases:
```
    @factory
    class Args:
        arg: Literal[0, 1, 2]

    args = parse_test(Args, ["--arg", "1"])
    assert args.arg == 1

    with raises(ParserError):
        args = parse_test(Args, ["--arg", "5"])
```

Some notes:
- A (shallow) copy of the dataclass field was required because mutating it directly would cause failures if multiple argument parsers are made: the first run would have an annotation like `Literal[0, 1, 2]`, but the second one would just have an `int` annotation. Went ahead and applied this copy strategy to the `Optional` field for consistency, though it's not strictly needed there.
- Currently assumes that all possible literals have the same type -- since there's no general `Union` support this seems reasonable to me.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -209,10 +210,8 @@ enum_test.py: error: argument --food: invalid choice: 'eggs' (choose from ['ham'
 
 **NOTE**: enums are passed by name on the command line and not by value.
 
-- compatibility with both `dataclass` and `attrs`
```
Comment (roee30) [inline README.md:212]: Why is this removed?

**Comment by roee30** [comment]
Thank you for your contribution. I only added a mutation of dataclass fields as an easy workaround. I do not wish to double down on it. Therefore, I will take the time to find a cleaner solution and then incorporate your contribution.

File: README.md (hunk 1 above)
Comment (brentyi) [reply README.md:212]: This line and the line below didn't make much sense to me here, so I assumed they were left by mistake.

Compatibility of the library with both dataclasses/attrs is already established above, as well as support for `field` and `attr.ib` arguments. Not sure why they'd be bullet points in a section about enums.

**Comment by brentyi** [comment]
Makes sense to me. Thanks @roee30!

**Comment by roee30** [comment]
Superseded by #15. Thanks @brentyi

**Comment by roee30** [comment]
Superseded by #15. Thanks @brentyi
